src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java

```diff
@@ -94,110 +95,36 @@
-     * in separate {@code -Xmsgs:} options, or in a comma-separated list
-     * for a single option. For example, the following are equivalent:
-     * <ul>
-     *     <li>{@code -Xmsgs:all} {@code -Xmsgs:-html}
-     *     <li>{@code -Xmsgs:all,-html}
-     * </ul>
-     *
-     * @param opts  options for doclint, derived from the corresponding doclet
-     *              command-line options
-     * @param customTagNames the names of custom tags, to be ignored by doclint
-     */
-    public void initDocLint(List<String> opts, Set<String> customTagNames) {
-        List<String> doclintOpts = new ArrayList<>();
-
-        // basic analysis of -Xmsgs and -Xmsgs: options to see if doclint is enabled
-        Set<String> groups = new HashSet<>();
-        boolean seenXmsgs = false;
-        for (String opt : opts) {
-            if (opt.equals(DocLint.XMSGS_OPTION)) {
-                groups.add("all");
-                seenXmsgs = true;
-            } else if (opt.startsWith(DocLint.XMSGS_CUSTOM_PREFIX)) {
-                String[] args = opt.substring(DocLint.XMSGS_CUSTOM_PREFIX.length())
-                        .split(DocLint.SEPARATOR);
-                for (String a : args) {
-                    if (a.equals("none")) {
-                        groups.clear();
-                    } else if (a.startsWith("-")) {
-                        groups.remove(a.substring(1));
-                    } else {
-                        groups.add(a);
-                    }
-                }
-                seenXmsgs = true;
-            }
-            doclintOpts.add(opt);
-        }
-
-        if (seenXmsgs) {
-            if (groups.isEmpty()) {
-                // no groups enabled; do not init doclint
-                return;
-            }
-        } else {
-            // no -Xmsgs options of any kind, use default
-            doclintOpts.add(DocLint.XMSGS_OPTION);
-        }
-
-        if (!customTagNames.isEmpty()) {
-            String customTags = String.join(DocLint.SEPARATOR, customTagNames);
-            doclintOpts.add(DocLint.XCUSTOM_TAGS_PREFIX + customTags);
-        }
-
-        doclintOpts.add(DocLint.XHTML_VERSION_PREFIX + "html5");
 
-        JavacTask t = BasicJavacTask.instance(toolEnv.context);
-        doclint = new DocLint();
-        doclint.init(t, doclintOpts.toArray(new String[0]), false);
-    }
+        elementUtils = configuration.docEnv.getElementUtils();
+        typeUtils = configuration.docEnv.getTypeUtils();
 
-    // TODO: fix this up correctly
-    public boolean haveDocLint() {
-        return (doclint == null);
+        // Note: this one use of DocEnvImpl is what prevents us tunnelling extra
+        // info from a doclet to its taglets via a doclet-specific subtype of
+        // DocletEnvironment.
+        toolEnv = ((DocEnvImpl)this.configuration.docEnv).toolEnv;
+        javacTypes = toolEnv.getTypes();
     }
 
     /*
      * TODO: This method exists because of a bug in javac which does not
```

Comment:
probably a good idea to include the bug ID here?

Reply:
@vicente-romero-oracle This relates to a discussion we should have elsewhere, separately, about the interactions between deprecation and packages, and whether we should support depicted packages. JLS does not list packages in the list of kinds of elements that can be deprecated, in which case this comment is probably incorrect, as the bug may actually be here in javadoc for trying to support deprecated packages!